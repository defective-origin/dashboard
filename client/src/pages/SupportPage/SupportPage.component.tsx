import React from 'react'

// ---| core |---
import { cn } from 'tools'

// ---| pages |---
import Page, { PageProps } from 'pages/Page'

// ---| screens |---
import SupportTable from 'screens/tables/SupportTable'

// ---| components |---
// ---| self |---
import css from './SupportPage.module.scss'

export type SupportPageProps = PageProps

/**
 * Component description.
 *
 * How to use
 * @example
 * <SupportPage />
 */
export function SupportPage(props: SupportPageProps): JSX.Element {
  const { children, className, ...otherProps } = props
  const _className = cn(css.SupportPage, className)

  return (
    <Page
      className={_className}
      name='PAGES.SUPPORT'
      menu={[{ start: 'add', tooltip: 'new request' }]}
      {...otherProps}
    >
      <Page.Content p='sm'>
        <SupportTable />

        {children}
      </Page.Content>
    </Page>
  )
}

SupportPage.displayName = 'SupportPage'

export default SupportPage
