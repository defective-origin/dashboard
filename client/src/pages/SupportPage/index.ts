/* INJECT_IMPORT_PLACE */

/* INJECT_DEFAULT_EXPORT_PLACE */
export { default } from './SupportPage.component'

/* INJECT_EXPORT_PLACE */
export * from './SupportPage.component'
