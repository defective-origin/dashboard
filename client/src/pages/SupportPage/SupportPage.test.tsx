// ---| tests |---
import { render } from 'tests'

// ---| self |---
import SupportPage from './SupportPage.component'

describe('[SupportPage] component', () => {
  it('should render component', () => {
    const container = render(<SupportPage />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})
