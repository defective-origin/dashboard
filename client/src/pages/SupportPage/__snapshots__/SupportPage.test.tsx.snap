// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[SupportPage] component > should render component 1`] = `
<DocumentFragment>
  <div
    class="_SupportPage_52d382"
  />
</DocumentFragment>
`;
